sdk/monitor/Azure.Monitory.Query/tests/LogsQueryClientClientLiveTests.cs

```diff
@@ -152,10 +153,227 @@ public async Task CanQueryBatch()
+        {
+            var client = CreateClient();
+
+            Response<LogsQueryResult> results = await client.QueryAsync(TestEnvironment.WorkspaceId,
+                $"datatable (DateTime: datetime, Bool:bool, Guid: guid, Int: int, Long:long, Double: double, String: string, Timespan: timespan, Decimal: decimal, NullBool: bool)" +
+                "[" +
+                "datetime(2015-12-31 23:59:59.9)," +
+                "false," +
+                "guid(74be27de-1e4e-49d9-b579-fe0b331d3642)," +
+                "12345," +
+                "1234567890123," +
+                "12345.6789," +
+                "\"string value\"," +
+                "10s," +
+                "decimal(0.10101)," +
+                "bool(null)" +
+                "]");
+
+            LogsQueryResultRow row = results.Value.PrimaryTable.Rows[0];
+
+            Assert.AreEqual(DateTimeOffset.Parse("2015-12-31 23:59:59.9+00:00"), row.GetDateTimeOffset("DateTime"));
+            Assert.AreEqual(DateTimeOffset.Parse("2015-12-31 23:59:59.9+00:00"), row.GetDateTimeOffset(0));
+            Assert.AreEqual("2015-12-31T23:59:59.9Z", row.GetObject("DateTime"));
+            Assert.AreEqual(false, row.GetBoolean("Bool"));
+            Assert.AreEqual(false, row.GetBoolean(1));
+            Assert.AreEqual(false, row.GetObject("Bool"));
+            Assert.AreEqual(Guid.Parse("74be27de-1e4e-49d9-b579-fe0b331d3642"), row.GetGuid("Guid"));
+            Assert.AreEqual(Guid.Parse("74be27de-1e4e-49d9-b579-fe0b331d3642"), row.GetGuid(2));
+            Assert.AreEqual("74be27de-1e4e-49d9-b579-fe0b331d3642", row.GetObject("Guid"));
+            Assert.AreEqual(12345, row.GetInt32("Int"));
+            Assert.AreEqual(12345, row.GetInt32(3));
+            Assert.AreEqual(12345, row.GetObject("Int"));
+            Assert.AreEqual(1234567890123, row.GetInt64("Long"));
+            Assert.AreEqual(1234567890123, row.GetInt64(4));
+            Assert.AreEqual(1234567890123, row.GetObject("Long"));
+            Assert.AreEqual(12345.6789d, row.GetDouble("Double"));
+            Assert.AreEqual(12345.6789d, row.GetDouble(5));
+            Assert.AreEqual(12345.6789d, row.GetObject("Double"));
+            Assert.AreEqual("string value", row.GetString("String"));
+            Assert.AreEqual("string value", row.GetString(6));
+            Assert.AreEqual("string value", row.GetObject("String"));
+            Assert.AreEqual(TimeSpan.FromSeconds(10), row.GetTimeSpan("Timespan"));
+            Assert.AreEqual(TimeSpan.FromSeconds(10), row.GetTimeSpan(7));
+            Assert.AreEqual("00:00:10", row.GetObject("Timespan"));
+            Assert.AreEqual(0.10101m, row.GetDecimal("Decimal"));
+            Assert.AreEqual(0.10101m, row.GetDecimal(8));
+            Assert.AreEqual("0.10101", row.GetObject("Decimal"));
+            Assert.True(row.IsNull("NullBool"));
+            Assert.True(row.IsNull(9));
+            Assert.IsNull(row.GetObject("NullBool"));
```

Comment:
We should do type asserts here also
ex.
`Assert.That(row.GetDateTimeOffset("DateTime"), Is.TypeOf(typeof(DateTimeOffset)));`

Reply:
GetDateTimeOffset return type is DateTimeOffset? What am I missing?

Reply:
True - I guess that is already asserting it. I was thinking we wanted to ensure it wasn't a boxed object. But that is happening higher up the stack.